### ISSUE TITLE
Code hangs in IDE but not in standalone Lua 5.3

### ISSUE THREAD

**Issue by raingloom** [pkulchenko/ZeroBraneStudio#539]
I'm testing a pre-processor I wrote with the following code:
```lua
local file = package.searchpath ( 'offline.test', package.path )
print ( file )
file = io.open ( file )
print ( file )
file = file:read '*a', file:close ()
print ( file )
file = require'raincoat.Code.PreProcessor'.preProcess ( file )
print ( file )
print ( load ( file ) )
```
This is so far the only minimal test case I found, even directly loading the processed code works.
If the last line is commented out or changed to `load ( file )`, it works as expected, but if it is left as is, it hangs. Always reproducible with both F6 run and F5(debug) run.

Setup: Lua 5.3 from Arch packages (64 bit)
ZBStudio's system config points lua53 to '/usr/bin/lua'
preprocessor in question is: https://github.com/raingloom/Raincoat/blob/master/Code/PreProcessor.lua
'test.lua' is under './offline/test.lua'

**Comment by pkulchenko** [comment]
@raingloom, thank you for the report; I'll try to reproduce on my Arch install. Few questions:

(1) does it also happen using Lua 5.3 shipped with ZBS?
(2) what's the content of `test.lua` file?
(3) what if you hardcode the content of the file in the third to last line (like `file = [[whatever content preProcess generated]]`) and keep only the last three lines? Do you still get the same issue?

**Comment by raingloom** [comment]
tl;dr: it fixed itself somehow, I'll try to reproduce it again after the upgrade completes
(1)I think it didn't have one (???)
I was getting an error about a missing file when I set my project to 5.3, instead of 5.2 or 5.1, that was the reason I had to tweak their paths in the global config.
I should note that I'm using an AUR package..... ok, just noticed that it's a bit outdated, the new one is currently being downloaded. I haven't run a full upgrade in about a week.
(2)test.lua:
```
#for i = 1, 10 do
  print "TEST"
#end
return "testing"
```
(3) This is so Murphy. It runs now and I can't seem to make it hang.
I'm relatively new to Arch, so I might have overlooked something obvious. Anyways, I am/was also going to comment on the core dumping problem, as I was getting them a lot, but even the ulimit trick is not helping.

**Comment by pkulchenko** [comment]
(1) what kind of error were you getting? Yes, you should definitely have Lua 5.3 included (for both 32bit and 64bit Linux versions); check `bin/linux/x64/lua53` and `bin/linux/x64/clibs53`. I recently upgraded Lua 5.3 to Lua 5.3.1 and tested on Ubuntu.
(3) So, you can't reproduce it anymore? Regarding the core dumps, what version are you using? There have been several crash-related issues fixed on Linux couple of months ago and I haven't seen any crash reports after that. Check the recently released v1.20 and see if it works without issues for you.

**Comment by pkulchenko** [comment]
@raingloom, closing this for now. Re-open if you have more information on this and can reproduce on the latest release.

**Comment by raingloom** [comment]
(1) Can't recall the exact wording but it was "errorcode 2", which turned out to be a missing file error after a quick search.

**Comment by pkulchenko** [comment]
yes, errorcode 2 usually means "File not found".
